### PR TITLE
Rate limiter. You can't post another message within 5 seconds. Fixes #161

### LINKED ISSUE #161: Rate limiter for messages

### PR DESCRIPTION
**Pull request by JeremybellEU**

### CODE REVIEW

**Comment by Azkle** [comment]
This one is to strict.
When I make a typo and want to do a rectification orso, it's not possible.
I would say 5 messages in 10 seconds as limit, then there is room for such things.

**Comment by Jaapp-** [comment]
I agree with @Azkle, 5 second timeouts are fit for forums, chats are meant to be used actively. I'd say a 10 second penalty for posting 10 messages in 10 seconds would be fit, something along those lines.

File: src/test/java/endtoend/ChatMessagesEndToEndTest.java [hunk 1]
```diff
@@ -39,7 +39,8 @@ public void canPostMultipleMessagesIdIncreases() {
     }
 
     @Test
-    public void canRetrieveMessagesAfterPosting() {
+    public void canRetrieveMessagesAfterPosting() throws InterruptedException {
+        Thread.sleep(10000);
```
Comment (JeremybellEU) [inline src/test/java/endtoend/ChatMessagesEndToEndTest.java]: Why is this needed?

File: src/main/java/me/moodcat/backend/rooms/RoomInstance.java [hunk 2]
```diff
@@ -273,10 +276,24 @@ public ChatMessageModel sendMessage(final ChatMessageModel model,
         return model;
     }
 
+    private void verifyNonSpamming(final User user) {
+        final long currentTime = System.currentTimeMillis();
```
Comment (JeremybellEU) [inline src/main/java/me/moodcat/backend/rooms/RoomInstance.java]: What happened with the System check? This has to be implemented for the EndToEndTests and general testing. Also with System messages they should not be filtered.

File: src/test/java/me/moodcat/backend/rooms/RoomInstanceTest.java [hunk 3]
```diff
@@ -55,17 +54,40 @@ public void setUp() {
...
-		    model.setMessage(String.valueOf(i));
-		    
-			instance.sendMessage(model, mock(User.class));
-		}
-
-		assertEquals(RoomInstance.MAXIMAL_NUMBER_OF_CHAT_MESSAGES, instance
-				.getMessages().size());
+    @Test
+    public void whenTooManyMessagesRemoveOneFromList() {
+
```
Comment (JeremybellEU) [inline src/test/java/me/moodcat/backend/rooms/RoomInstanceTest.java]: This whitespace should be removed.

File: src/test/java/me/moodcat/backend/rooms/RoomInstanceTest.java [hunk 4]
```diff
@@ -55,17 +54,40 @@ public void setUp() {
...
+	@Test(expected = IllegalArgumentException.class)
+	public void whenOneUserSendsMessagesTooFastThrowsException() {
+	    ChatMessageModel model = new ChatMessageModel();
+        model.setMessage("Spam");
+        
+
+        User user = mock(User.class);
+        when(user.getId()).thenReturn(1337);
+	    
+	    instance.sendMessage(model, user);
```
Comment (JeremybellEU) [inline src/test/java/me/moodcat/backend/rooms/RoomInstanceTest.java]: This should be put in a for-loop to increase maintainability and readibility.

**Comment by JeremybellEU** [comment]
Small remarks. After they are resolved, it can be merged.

File: src/main/java/me/moodcat/backend/rooms/RoomInstance.java (hunk 2 above)
Comment (Azkle) [inline src/main/java/me/moodcat/backend/rooms/RoomInstance.java]: I was doubting if that was needed and @JWGmeligMeyling said it wasn't needed and shoud be deleted.
Therefore I deleted that part.
